### PR TITLE
TDL-20794: Upgrade SDK Version and add missing fields.

### PR DESCRIPTION
**Pull request by hpatel41**
# Description of change
TDL-20794: Upgrade SDK Version and add missing fields.
- Upgraded the python-asana SDK version to **1.0.0**
- Added missing fields in the schema as per the Stability Analysis.

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_asana/schemas/projects.json [hunk 1]
```diff
@@ -53,8 +53,115 @@
     "current_status": {
       "type": [
         "null",
-        "string"
-      ]
+        "object"
```
Comment (dbshah1212) [inline tap_asana/schemas/projects.json]: Any reason for changing the type? It will lead to a Major release.

Comment (hpatel41) [reply tap_asana/schemas/projects.json]: Reverted the updating of datatype.

**Review by savan-chovatiya — APPROVED** [review]

**Review by dbshah1212 — APPROVED** [review]

**Comment by TaylorFacen-AL** [comment]
Looks like this PR added in a section column for the tasks table. Any idea on when this can get merged in @hpatel41 @dbshah1212 @savan-chovatiya ?

File: setup.py [hunk 2]
```diff
@@ -10,7 +10,7 @@
     classifiers=["Programming Language :: Python :: 3 :: Only"],
     py_modules=["tap_asana"],
     install_requires=[
-        "asana==0.10.2",
+        "asana==1.0.0",
```
Comment (RushiT0122) [inline setup.py]: Latest `SDK version 3.0.0` is available.

File: setup.py [hunk 3]
```diff
@@ -10,7 +10,7 @@
...
         'singer-python==5.12.2'
```
Comment (RushiT0122) [inline setup.py]: Latest `singer-python version 5.13.0` available

**Review by RushiT0122 — CHANGES_REQUESTED** [review]

**Review by RushiT0122 — APPROVED** [review]

**Comment by rdeshmukh15** [comment]
Change the `Description of change` for this PR

File: tap_asana/schemas/portfolios.json [hunk 4]
```diff
@@ -271,6 +271,138 @@
...
+                      "null",
+                      "string"
+                    ]
+                  },
+                  "date_time": {
+                    "type": [
+                      "null",
+                      "string"
+                    ]
+                  }
```
Comment (rdeshmukh15) [inline tap_asana/schemas/portfolios.json]: Mention the format for these fields

File: tap_asana/schemas/portfolios.json [hunk 5]
```diff
@@ -480,6 +612,264 @@
...
+                  "null",
+                  "string"
+                ]
+              },
+              "date_time": {
+                "type": [
+                  "null",
+                  "string"
+                ]
+              }
```
Comment (rdeshmukh15) [inline tap_asana/schemas/portfolios.json:691]: Mention the date-time format here too

File: tap_asana/schemas/projects.json [hunk 6]
```diff
@@ -221,6 +221,78 @@
...
+                  "null",
+                  "string"
+                ]
+              },
+              "date_time": {
+                "type": [
+                  "null",
+                  "string"
+                ]
+              }
```
Comment (rdeshmukh15) [inline tap_asana/schemas/projects.json]: Add the date-time format wherever required

**Review by rdeshmukh15 — COMMENTED** [review]
Add the new fields introduced as per the stability analysis

**Comment by rdeshmukh15** [comment]
If possible can you please fetch `fields` list dynamically(i.e. from the catalog)  instead of hardcoding as defined in every file of streams.

File: tap_asana/schemas/portfolios.json (hunk 4 above)
Comment (rdeshmukh15) [reply tap_asana/schemas/portfolios.json]: As per the documentation, it is supposed to be the date and date_time with string datatype and hence, not added the format but can add the format date-time for field due_on

File: tap_asana/schemas/projects.json (hunk 6 above)
Comment (rdeshmukh15) [reply tap_asana/schemas/projects.json]: As per the documentation, it is supposed to be the date and date_time with string datatype and hence, not added the format but can add the format date-time for fields due_on and due_date

**Review by nitingaikwad1 — APPROVED** [review]

**Review by kethan1122 — APPROVED** [review]

**Pull request merged**